zsh/zshenv.symlink

```diff
@@ -1,21 +1,72 @@
-# path to dotfiles
-export DOTFILES="$HOME/.odotfiles"
-
-# `brew --prefix` takes ~25ms on my machine (to see how long it takes on yours
-# try `hyperfine 'brew --prefix'`). To avoid that overhead I hardcode the
-# path (seems pretty stable).
-export HOMEBREW_PREFIX="/usr/local"
-
-export GOPATH="$HOME/go"
-
-export VAULT_ADDR=https://vault.services.opendoor.com:8200
-
-# colored paging (enables colored man pages)
-# ref - https://github.com/sorin-ionescu/prezto/blob/cfeb8cd6c9f60c2a928f4c706d0137c7de2ef106/modules/environment/init.zsh#L59-L67
-export LESS_TERMCAP_mb=$'\E[01;31m'      # begins blinking
-export LESS_TERMCAP_md=$'\E[01;31m'      # begins bold
-export LESS_TERMCAP_me=$'\E[0m'          # ends mode
-export LESS_TERMCAP_se=$'\E[0m'          # ends standout-mode
-export LESS_TERMCAP_so=$'\E[00;47;30m'   # begins standout-mode
-export LESS_TERMCAP_ue=$'\E[0m'          # ends underline
-export LESS_TERMCAP_us=$'\E[01;32m'      # begins underline
+main() {
+  # set DOTFILES env var (statically if possible for speed, else dynamically)
+  if [ -d "$HOME/.files" ]; then
+   # to avoid unnecessary overhead assume the `$HOME/.files` directory contains
+   # this config if it exists
+    export DOTFILES="$HOME/.files"
+  else
+    dynamically_determine_dotfiles # see function at end of file
+  fi
+
+  # `brew --prefix` takes ~25ms on my machine (to see how long it takes on yours
+  # try `hyperfine 'brew --prefix'`). To avoid that overhead I hardcode the
+  # path (seems pretty stable).
+  [ -d /usr/local/Homebrew ] && export HOMEBREW_PREFIX="/usr/local"
+
+  export GOPATH="$HOME/go"
+
+  export VAULT_ADDR=https://vault.services.opendoor.com:8200
+
+  # output ANSI "color" escape sequences
+  # value of LESS is passed to `less` as args
+  export LESS=" -R"
+  # LESSOPEN configures preprocessor that colors ANSI escape sequences
+  # http://www.andre-simon.de/doku/highlight/en/integration.php
+  export LESSOPEN="| $(command -v highlight) %s --out-format truecolor \
+--line-numbers --force --style base16/onedark --no-trailing-nl"
```

Comment:
@sbarbeau-od `less` syntax highlighting using [`highlight`](http://www.andre-simon.de/doku/highlight/en/highlight.php)

Reply:
the update regarding highlighting in less works in the same terminal after a `make setup`, but it doesn't work if I open a new terminal and I have the following issue for every files:
``` less file.py
zsh:1: permission denied: file.py```

Reply:
if I manually export LESS and LESSOPEN in my current terminal it works, but not if I open a new terminal 🤔